### DISCUSSION TITLE
Allow to import 'Sequence' from 'typing' rather than from 'collections.abc'

### DISCUSSION THREAD

**Discussion by p-j-smith** [astral-sh/ruff#3383] [Q&A] [answered]
Hi, thanks for making `ruff`! It's awesome.

I'm having some trouble with ruff modifying my imports and I can't figure out which rule I need to disable.

I would like to import `Sequence` from `typing` for use with Python 3.8. However, when I run ruff on the file, the imports are changed such that `Sequence` is imported from `collections.abc` instead. This then breaks my code on Python 3.8 with the following error:

```
TypeError: 'ABCMeta' object is not subscriptable
```

### Minimal example
Here's what I would like to do:
```python
from typing import Sequence, Callable, Float, Union

FloatOrFloats = Union[Float, Sequence[Float]]
```

but ruff re-formats this to:
```python
from collections.abc import Sequence
from typing import Float, Union

FloatOrFloats = Union[Float, Sequence[Float]]

```

Any help on how I can prevent the imports from changing would be much appreciated!

**Comment by p-j-smith** [comment]
Sorry I just realised that you can run ruff with a `--show-fixes` flag that will tell you what rules led to the changes. Running this helped me realise I can prevent the import from changing like this:

```python
from typing import Sequence, Float, Union  # noqa: UP035

FloatOrFloats = Union[Float, Sequence[Float]]

```

**Reply by charliermarsh** [reply]
A more "sustainable" option might be to set your [`target-version`](https://beta.ruff.rs/docs/settings/#target-version) to `py38`, like:

```toml
[tool.ruff]
target-version = "py38"
```

That should avoid any rewrites that aren't compatible with Python 3.8!

**Reply by p-j-smith** [reply]
> A more "sustainable" option might be to set your [target-version](https://beta.ruff.rs/docs/settings/#target-version) to py38

that's a **much** better solution, thanks!

**Reply by charliermarsh** [reply]
(Re-commented the same thing below to "Mark as answer", I'm new to Discussions, learning as I go :))

**Reply by p-j-smith** [reply]
perfect, I was wondering how to mark your comment as an answer

**Comment by charliermarsh** [comment, marked as answer]
A more "sustainable" option might be to set your [`target-version`](https://beta.ruff.rs/docs/settings/#target-version) to `py38`, like:

```toml
[tool.ruff]
target-version = "py38"
```

That should avoid any rewrites that aren't compatible with Python 3.8!
